screens/NewAccount/NewAccountPasskeyScreen.tsx

```diff
@@ -0,0 +1,334 @@
+import React, { memo, useCallback, useRef } from "react";
+
+import { NewAccountScreenComp } from "@components/NewAccount/NewAccountScreenComp";
+import { NewAccountPictoTitleSubtitle } from "@components/NewAccount/NewAccountTitleSubtitlePicto";
+import {
+  PasskeyAuthStoreProvider,
+  usePasskeyAuthStoreContext,
+} from "@features/onboarding/passkey/passkeyAuthStore";
+import { translate } from "@i18n";
+import { useRouter } from "../../navigation/useNavigation";
+import { isMissingConverseProfile } from "../Onboarding/Onboarding.utils";
+import { addWalletToPasskey } from "@/utils/passkeys/add-wallet-to-passkey";
+import { initXmtpClientFromViemAccount } from "@/components/Onboarding/init-xmtp-client";
+import { Button } from "@/design-system/Button/Button";
+import { TextField } from "@/design-system/TextField/TextField";
+import { Text } from "@/design-system/Text";
+import { createPasskey } from "@/utils/passkeys/create-passkey";
+import { loadAccountFromPasskey } from "@/utils/passkeys/load-client-from-passkey";
+import { captureErrorWithToast } from "@/utils/capture-error";
+import logger from "@/utils/logger";
+import { converseEventEmitter } from "@/utils/events";
+
+export const NewAccountPasskeyScreen = memo(function () {
+  return (
+    <PasskeyAuthStoreProvider>
+      <Content />
+    </PasskeyAuthStoreProvider>
+  );
+});
+
+const Content = memo(function Content() {
+  const router = useRouter();
+
+  const loading = usePasskeyAuthStoreContext((state) => state.loading);
+
+  const statusString = usePasskeyAuthStoreContext(
```

Comment:
instead of selecting all these values, can we just destructure the store context returned here? Can't be hurting perf that much